### DISCUSSION TITLE
LinkWidget - adding attrs to choices

### DISCUSSION THREAD

**Discussion by mflpopescu** [carltongibson/django-filter#1503] [Q&A]
I figured out additional attributes can be added to the LinkWidget using 
```
LinkWidget(attrs={'class': 'list-group'}))
```
Are there any options though for adding extra attributes to the choices ? For example in the case of 'bootstrap list-group' classes need to be added to the **ul** and **li** elements and using the syntax above I can only add a class to the **ul** element

https://getbootstrap.com/docs/4.0/components/list-group/

```
<ul id="id_manufacturer" class="list-group">
<li><a href="?manufacturer=">All</a></li>
<li><a href="?manufacturer=Borbet">Borbet</a></li>
<li><a class="selected" href="?manufacturer=Momo">Momo</a></li>
</ul>
```

filters.py
```
import django_filters
from .models import *

class AlloyFilter(django_filters.FilterSet):
    manufacturer = django_filters.ChoiceFilter(
        choices=[],
        label='Manufacturer',
        widget = django_filters.widgets.LinkWidget(attrs={'class': 'list-group'}))

    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)
        """
        explicitly providing the choices
        """
        self.filters['manufacturer'].extra['choices'] = [
            (wheel, wheel)
            for wheel in WheelItem.objects.values_list('manufacturer', flat=True).distinct().order_by('manufacturer')
        ]

```

**Comment by carltongibson** [comment]
Did you see the docs for `ModelChoiceIterator` and `ModelChoiceIteratorValue`? These allow you to customise the options generation. https://docs.djangoproject.com/en/4.0/ref/forms/fields/#modelchoiceiterator
